extensions/cloud/portability-cloud-microsoft/src/main/java/org/dataportabilityproject/cloud/microsoft/cosmos/TableStoreConfiguration.java

```diff
@@ -0,0 +1,79 @@
+package org.dataportabilityproject.cloud.microsoft.cosmos;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+
+/** Configuration required to initialize the Azure job store */
+public class TableStoreConfiguration {
+  private String accountName;
+  private String accountKey;
+  private String blobKey;
+  private String partitionKey;
+  private ObjectMapper mapper;
+
+  public String getAccountName() {
+    return accountName;
+  }
+
+  public String getAccountKey() {
+    return accountKey;
+  }
+
+  public String getPartitionKey() {
+    return partitionKey;
+  }
+
+  public String getBlobKey() {
+    return blobKey;
+  }
+
+  public ObjectMapper getMapper() {
+    return mapper;
+  }
+
+  public static class Builder {
```

Comment:
FYI, totally optional but if you are interested, AutoValue does builders pretty easily (see JobAuthorization for an example)

Reply:
I'll take a look. One thing is the autovalue annotation plugin is not playing nicely with IDEA's incremental compilation and Gradle since multiple generated output files are resulting in duplicates. I think this may be a bug in IDEA but haven't had time to look.

Reply:
OK thx.